### PR TITLE
Fixes #276 - Unknown configuration options should throw an error

### LINKED ISSUE #276: Unknown configuration options should throw an error
Example, if you use `featureFlag` instead of `featureFlags`:

```yaml
generate:
  - types
featureFlag:
  - nullableSchemas
```

This is easy to do because the CLI is `--feature-flag` (taking multiple).
 
We already print the following to stdout:

```none
Swift OpenAPI Generator is running with the following configuration:
- OpenAPI document path: ...
- Configuration path: ...
- Generator modes: types, client
- Feature flags: ...
- Output file names: Types.swift, Client.swift
- Output directory: ...
- Diagnostics output path: <none - logs to stderr>
- Current directory: ...
- Plugin source: <none>
- Is dry run: false
- Additional imports: <none>
```

Probably should print something here about unknown keys?

### PR DESCRIPTION
**Pull request by PARAIPAN9**
### Motivation

Fixes #276

### Modifications

Create a function that extracts the keys from the config file and compares them with the UserConfig Coding Keys

### Result

When the config file loads, it also checks for the keys to match

### Test Plan

Loaded different config files with different configurations to see if it works.

### CODE REVIEW

File: Sources/swift-openapi-generator/GenerateOptions.swift [hunk 1]
```diff
@@ -99,6 +117,16 @@ extension _GenerateOptions {
         }
         do {
             let data = try Data(contentsOf: config)
+            let configAsString = String(data: data, encoding: .utf8)
+
+            let yamlKeys = extractYamlKeys(yamlString: configAsString!)
```
Comment (czechboy0) [inline Sources/swift-openapi-generator/GenerateOptions.swift:122]:
```suggestion
            let configAsString = String(decoding: data, as: UTF8.self)
            let yamlKeys = extractYamlKeys(yamlString: configAsString)
```

File: Sources/swift-openapi-generator/UserConfig.swift [hunk 2]
```diff
@@ -29,4 +29,12 @@ struct _UserConfig: Codable {
 
     /// A set of features to explicitly enable.
     var featureFlags: FeatureFlags?
+
+    static let codingKeysRawValues = Set(CodingKeys.allCases.map({ $0.rawValue }))
```
Comment (czechboy0) [inline Sources/swift-openapi-generator/UserConfig.swift:33]: Please add a comment for the symbol.

File: Sources/swift-openapi-generator/GenerateOptions.swift [hunk 3]
```diff
@@ -86,7 +86,25 @@ extension _GenerateOptions {
...
+
+        var extractedKeys = [String]()
+
+        for line in lines {
+            if let range = line.range(of: ":") {
+                let key = line[line.startIndex..<range.lowerBound].trimmingCharacters(in: .whitespaces)
+                extractedKeys.append(key)
+            }
+        }
+        return extractedKeys
```
Comment (czechboy0) [inline Sources/swift-openapi-generator/GenerateOptions.swift:107]: I agree we need to extract the keys somehow, but I don't think we should try to write our own YAML parser here. Can you see if you can use the lower level Yams API to do the same? https://www.jpsim.com/Yams/Classes/Parser.html

**Review by czechboy0 — CHANGES_REQUESTED** [review]

File: Sources/swift-openapi-generator/GenerateOptions.swift (hunk 3 above)
Comment (PARAIPAN9) [reply Sources/swift-openapi-generator/GenerateOptions.swift:107]: @czechboy0, thank you for the comments. I wasn't sure about the YAML parsing method, which is why I opened the PR without writing tests for it. I will check the link provided by you to explore other possibilities.

File: Sources/swift-openapi-generator/GenerateOptions.swift [hunk 4]
```diff
@@ -86,7 +86,42 @@ extension _GenerateOptions {
...
+            let parser = try Parser(yaml: yamlString)
+            if let rootNode = try parser.singleRoot(),
+                case let .mapping(mapping) = rootNode
+            {
+                for (key, _) in mapping {
+                    yamlKeys.append(key.string ?? "")
+                }
+            }
+        } catch {
+            print("Error parsing YAML: \(error)")
```
Comment (czechboy0) [inline Sources/swift-openapi-generator/GenerateOptions.swift:109]: I think it's better to just let the error be thrown one level up, and there emit a ValidationError saying the config isn't valid and include the description of the error thrown by Yams. So you can remove the do/catch here.

**Review by czechboy0 — CHANGES_REQUESTED** [review]
Just one more nit, and can you add a unit test please?

File: Sources/swift-openapi-generator/GenerateOptions.swift (hunk 4 above)
Comment (PARAIPAN9) [reply Sources/swift-openapi-generator/GenerateOptions.swift:109]: Sure thing!!

Comment (PARAIPAN9) [reply Sources/swift-openapi-generator/GenerateOptions.swift:109]: Just a quick question, where would be a good place in the codebase, to write the unit test?

Comment (czechboy0) [reply Sources/swift-openapi-generator/GenerateOptions.swift:109]: You could move the logic close to the YAML parser in the core module, and that one has unit tests that you can add to. We don't yet have a unit test target for the CLI module.

Comment (PARAIPAN9) [reply Sources/swift-openapi-generator/GenerateOptions.swift:109]: perfect, TY!!

**Comment by PARAIPAN9** [comment]
Please feel free to comment, suggest changes if still something is not right or could be improved. I really appreciate your input!!

**Review by czechboy0 — APPROVED** [review]
Thanks!

**Pull request merged**